### PR TITLE
allow for event handlers to ignore args

### PR DESCRIPTION
**Pull request by adhami3310**

### CODE REVIEW

**Review by masenf — APPROVED** [review]
looks good aside from the large amount of commented out test code; should this just be removed to avoid confusion?

**Comment by adhami3310** [comment]
> should this just be removed to avoid confusion?

i added them to valid test cases instead

File: reflex/event.py [hunk 1]
```diff
@@ -1058,8 +1058,8 @@ def get_hydrate_event(state) -> str:
 
 
 def call_event_handler(
-    event_handler: EventHandler | EventSpec,
-    arg_spec: ArgsSpec | Sequence[ArgsSpec],
+    event_callback: EventHandler | EventSpec,
+    event_spec: ArgsSpec | Sequence[ArgsSpec],
```
Comment (Lendemor) [inline reflex/event.py:1062]: Won't it be confusing to have 
`event_callback` of type `EventSpec`
but then
`event_spec` of type `ArgsSpec` ?

Tho I'm not sure what we could use instead, so maybe it's fine.

**Review by Lendemor — DISMISSED** [review]

File: reflex/event.py (hunk 1 above)
Comment (adhami3310) [reply reflex/event.py:1062]: I would love to change existing terminology but that's the best i could do while working in the system :/

**Review by Lendemor — APPROVED** [review]

**Pull request merged**